pages/en/lb3/Advanced-topics-data-sources.md

```diff
@@ -35,6 +32,9 @@ It's a factory for data access logic around model classes. With the ability to p
 
 ## Creating a DataSource programmatically
 
+{% include warning.html content="Don't put production database credentials explicitly in JSON or JavaScript files, where they could be a security vulnerability.  Instead, load the values from environment variables.  For more information, see [Specifying database credentials with environment variables](Attaching-models-to-data-sources.html#specifying-database-credentials-with-environment-variables)
+"%}
+
```

Comment:
ok for the warning but it stands that the problem expressed [in this comment](https://github.com/strongloop/loopback.io/pull/317#issuecomment-285854277) is still not covered by the doc.

The doc explains how to create datasource programmatically but some info should be added on where to use this snippet, and how to connect models on these datasources (programmatically I suppose, but one could assume that they could reference datasources created programmatically in the model-config.json